server/functions/framework/interpreter_operation.go

```diff
@@ -0,0 +1,77 @@
+	OpCode_Get                        // https://www.postgresql.org/docs/15/plpgsql-statements.html#PLPGSQL-STATEMENTS-DIAGNOSTICS
+	OpCode_If                         // https://www.postgresql.org/docs/15/plpgsql-control-structures.html#PLPGSQL-CONDITIONALS
+	OpCode_InsertInto                 // https://www.postgresql.org/docs/15/plpgsql-statements.html
+	OpCode_Loop                       // https://www.postgresql.org/docs/15/plpgsql-control-structures.html#PLPGSQL-CONTROL-STRUCTURES-LOOPS
+	OpCode_Perform                    // https://www.postgresql.org/docs/15/plpgsql-statements.html
+	OpCode_Query                      // This is just a standard query, nothing special
+	OpCode_Return                     // https://www.postgresql.org/docs/15/plpgsql-control-structures.html#PLPGSQL-STATEMENTS-RETURNING
+	OpCode_ScopeBegin                 // This is used for scope control, specific to Doltgres
+	OpCode_ScopeEnd                   // This is used for scope control, specific to Doltgres
+	OpCode_ScopeRestart               // This is used for scope control, specific to Doltgres
+	OpCode_SelectInto                 // https://www.postgresql.org/docs/15/plpgsql-statements.html
+	OpCode_When                       // https://www.postgresql.org/docs/15/plpgsql-control-structures.html#PLPGSQL-CONDITIONALS
+	OpCode_While                      // https://www.postgresql.org/docs/15/plpgsql-control-structures.html#PLPGSQL-CONTROL-STRUCTURES-LOOPS
+	OpCode_UpdateInto                 // https://www.postgresql.org/docs/15/plpgsql-statements.html
+)
+
+// IterationType represents how iteration should proceed. Each type has a different ruleset that it follows, and all
+// forms of control flow can be represented by one of these types.
+type IterationType uint8
+
+const (
+	// IterationType_Normal :: Every statement encountered should be executed. If a conditional continuation is
+	// encountered (ELSIF or ELSE), then this turns into IterationType_SkipToEnd.
+	IterationType_Normal IterationType = iota
+	// IterationType_SkipToConditional :: Every statement should be ignored until a conditional is encountered. Only
+	// conditionals with the iterationID will be considered. This reverts to IterationType_Normal if a ScopeEnd with the
+	// iterationID is encountered.
+	IterationType_SkipToConditional
+	// IterationType_SkipToRestart :: Every statement is ignored until a ScopeRestart or ScopeEnd with the iterationID
+	// is encountered.
+	IterationType_SkipToRestart
+	// IterationType_SkipToEnd :: Every statement is ignored until a ScopeEnd with the iterationID is encountered.
+	IterationType_SkipToEnd
+	// IterationType_Reverse :: Every statement is ignored until a ScopeBegin or Declare with the iterationID is encountered.
+	IterationType_Reverse
+)
+
+// InterpreterOperation is an operation that will be performed by the interpreter.
+type InterpreterOperation struct {
+	OpCode
```

Comment:
Is there a reason to embed this instead of using a normal field?

Reply:
Nope, I was just going to name the field `OpCode` so embedding does it automatically. No reason otherwise.